File: backend/src/main/java/com/brewnow/entity/Address.java
package com.brewnow.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 收货地址实体
 */
@Data
public class Address {
    private Integer addressId;
    private Integer userId;
    private String receiverName;
    private String contactPhone;
    private String province;
    private String city;
    private String district;
    private String detailAddress;
    private String tag;
    private Boolean isDefault;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
    private LocalDateTime deletedAt;
}

File: frontend/src/api/address.ts
import request from './request'

export interface UserAddress {
  addressId?: number
  userId?: number
  receiverName: string
  contactPhone: string
  province?: string
  city?: string
  district?: string
  detailAddress: string
  tag?: string
  isDefault?: boolean
  createTime?: string
  updateTime?: string
}

export const addressApi = {
  getList() {
    return request.get<UserAddress[]>('/address/list')
  },
  getDefault() {
    return request.get<UserAddress | null>('/address/default')
  },
  addAddress(data: UserAddress) {
    return request.post<UserAddress>('/address/add', data)
  },
  quickAddAddress(data: UserAddress) {
    return request.post<UserAddress>('/address/quick-add', data)
  },
  updateAddress(data: UserAddress) {
    return request.put('/address/update', data)
  },
  deleteAddress(addressId: number) {
    return request.delete(`/address/${addressId}`)
  },
  setDefault(addressId: number) {
    return request.put(`/address/default/${addressId}`)
  }
}

